a short tests without valid credetials shows a traceback which should be catched.

~~~
* including generated source, if any, saving to: /home/resmo/.ansible_module_generated
* ansiballz module detected; extracted module source to: /home/resmo/debug_dir
***********************************
RAW OUTPUT

Traceback (most recent call last):
  File "/home/resmo/debug_dir/ansible_module_jenkins_job_facts.py", line 190, in <module>
    main()
  File "/home/resmo/debug_dir/ansible_module_jenkins_job_facts.py", line 186, in main
    run_module()
  File "/home/resmo/debug_dir/ansible_module_jenkins_job_facts.py", line 168, in run_module
    all_jobs = jenkins_conn.get_all_jobs()
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 814, in get_all_jobs
    jobs = [(0, "", self.get_info(query=JOBS_QUERY)['jobs'])]
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 551, in get_info
    Request(self._build_url(url))
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 430, in jenkins_open
    self.maybe_add_crumb(req)
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 303, in maybe_add_crumb
    self._build_url(CRUMB_URL)), add_crumb=False)
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 461, in jenkins_open
    raise JenkinsException('Error in request: %s' % (e.reason))
jenkins.JenkinsException: Error in request: [Errno 111] Connection refused
~~~

--- a/lib/ansible/modules/web_infrastructure/jenkins_job_facts.py
+++ b/lib/ansible/modules/web_infrastructure/jenkins_job_facts.py
@@ -108,7 +108,11 @@ from ansible.module_utils.basic import AnsibleModule
 from ansible.module_utils._text import to_native
 
 
-def get_jenkins_connection(url, username, password=None, token=None):
+def get_jenkins_connection(module):
+    url = module.params["url"]
+    username = module.params.get("user")
+    password = module.params.get("password")
+    token = module.params.get("token")
     if username and (password or token):
         return jenkins.Jenkins(url, username, password or token)
     elif username:
@@ -124,34 +128,8 @@ def test_dependencies(module):
             "see http://python-jenkins.readthedocs.io/en/latest/install.html")
 
 
-def run_module():
-    module = AnsibleModule(
-        argument_spec=dict(
-            name=dict(type='str', required=False),
-            glob=dict(type='str', required=False),
-            password=dict(required=False, no_log=True),
-            token=dict(required=False, no_log=True),
-            url=dict(required=False, default="http://localhost:8080"),
-            user=dict(required=False)),
-        mutually_exclusive=[
-            ['password', 'token'],
-            ['name', 'glob'],
-        ],
-        supports_check_mode=True)
-
-    test_dependencies(module)
-
-    try:
-        jenkins_conn = get_jenkins_connection(
-            module.params["url"],
-            module.params.get("user"),
-            password=module.params.get("password"),
-            token=module.params.get("token"))
-    except Exception as err:
-        module.fail_json(
-            msg='Unable to connect to Jenkins server, %s' % to_native(err),
-            exception=traceback.format_exc())
-
+def get_jobs(module):
+    jenkins_conn = get_jenkins_connection(module)
     jobs = []
     if module.params.get("name"):
         try:
@@ -176,6 +154,32 @@ def run_module():
         for job in jobs:
             if "_class" in job:
                 del job["_class"]
+    return jobs
+
+
+def run_module():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str', required=False),
+            glob=dict(type='str', required=False),
+            password=dict(required=False, no_log=True),
+            token=dict(required=False, no_log=True),
+            url=dict(required=False, default="http://localhost:8080"),
+            user=dict(required=False)),
+        mutually_exclusive=[
+            ['password', 'token'],
+            ['name', 'glob'],
+        ],
+        supports_check_mode=True)
+
+    test_dependencies(module)
+
+    try:
+        jobs = get_jobs(module)
+    except jenkins.JenkinsException as err:
+        module.fail_json(
+            msg='Unable to connect to Jenkins server, %s' % to_native(err),
+            exception=traceback.format_exc())
 
     # in the event of a successful module execution, you will want to
     # simple AnsibleModule.exit_json(), passing the key/value results

--- a/lib/ansible/modules/web_infrastructure/jenkins_job_facts.py
+++ b/lib/ansible/modules/web_infrastructure/jenkins_job_facts.py
@@ -152,23 +152,28 @@ def get_jobs(module):
     if module.params.get("name"):
         try:
             job_info = jenkins_conn.get_job_info(module.params.get("name"))
+        except jenkins.NotFoundException:
+            pass
+        else:
             jobs.append({
                 "name": job_info["name"],
                 "fullname": job_info["fullName"],
                 "url": job_info["url"],
                 "color": job_info["color"]
             })
-        except jenkins.NotFoundException:
-            pass
+
     else:
         all_jobs = jenkins_conn.get_all_jobs()
         if module.params.get("glob"):
-            jobs.extend([
+            jobs.extend(
                 j for j in all_jobs
-                if fnmatch.fnmatch(j["fullname"], module.params.get("glob"))
-            ])
+                if fnmatch.fnmatch(j["fullname"], module.params.get("glob")))
         else:
             jobs = all_jobs
+        # python-jenkins includes the internal Jenkins class used for each job
+        # in its return value; we strip that out because the leading _ (and the
+        # fact that it's not documented in the python-jenkins docs) indicates
+        # that it's not part of the dependable public interface.
         for job in jobs:
             if "_class" in job:
                 del job["_class"]
@@ -207,9 +212,5 @@ def run_module():
     module.exit_json(changed=False, jobs=jobs)
 
 
-def main():
-    run_module()
-
-
 if __name__ == '__main__':
-    main()
+    run_module()

--- a/lib/ansible/modules/web_infrastructure/jenkins_job_facts.py
+++ b/lib/ansible/modules/web_infrastructure/jenkins_job_facts.py
@@ -29,6 +29,9 @@ options:
   glob:
     description:
       - A shell glob of Jenkins job names to fetch facts about.
+  color:
+    description:
+      - Only fetch jobs with the given status color.
   password:
     description:
       - Password to authenticate with the Jenkins server.
@@ -81,6 +84,21 @@ EXAMPLES = '''
 # Get facts about jobs matching a shell glob using basic auth
 - jenkins_job_facts:
     glob: some-job-*
+    user: admin
+    password: hunter2
+  register: my_jenkins_job_facts
+
+# Get facts about all failing jobs using basic auth
+- jenkins_job_facts:
+    color: red
+    user: admin
+    password: hunter2
+  register: my_jenkins_job_facts
+
+# Get facts about passing jobs matching a shell glob using basic auth
+- jenkins_job_facts:
+    name: some-job-*
+    color: blue
     user: admin
     password: hunter2
   register: my_jenkins_job_facts
@@ -154,6 +172,10 @@ def get_jobs(module):
         for job in jobs:
             if "_class" in job:
                 del job["_class"]
+
+    if module.params.get("color"):
+        jobs = [j for j in jobs if j["color"] == module.params.get("color")]
+
     return jobs
 
 
@@ -162,6 +184,7 @@ def run_module():
         argument_spec=dict(
             name=dict(type='str', required=False),
             glob=dict(type='str', required=False),
+            color=dict(type='str', required=False),
             password=dict(required=False, no_log=True),
             token=dict(required=False, no_log=True),
             url=dict(required=False, default="http://localhost:8080"),
@@ -181,8 +204,6 @@ def run_module():
             msg='Unable to connect to Jenkins server, %s' % to_native(err),
             exception=traceback.format_exc())
 
-    # in the event of a successful module execution, you will want to
-    # simple AnsibleModule.exit_json(), passing the key/value results
     module.exit_json(changed=False, jobs=jobs)
 
 

